python/llm/example/GPU/HF-Transformers-AutoModels/Model/phi-3-vision/generate.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import time
+import torch
+import argparse
+import requests
+
+from PIL import Image
+from ipex_llm.transformers import AutoModelForCausalLM
+from transformers import AutoProcessor
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Predict Tokens using `generate()` API for phi-3 model')
+    parser.add_argument('--repo-id-or-model-path', type=str, default="microsoft/Phi-3-vision-128k-instruct",
+                        help='The huggingface repo id for the phi-3-vision model to be downloaded'
+                             ', or the path to the huggingface checkpoint folder')
+    parser.add_argument('--image-url-or-path', type=str,
+                        default="http://farm6.staticflickr.com/5268/5602445367_3504763978_z.jpg",
+                        help='The URL or path to the image to infer')
+    parser.add_argument('--prompt', type=str, default="What is in the image?",
+                        help='Prompt to infer')
+    parser.add_argument('--n-predict', type=int, default=32,
+                        help='Max tokens to predict')
+
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+    image_path = args.image_url_or_path
+
+    # Load model in INT8,
+    # which convert the relevant layers in the model into INT8 format
+    # We here use INT8 instead of INT4 for better output
+    # `_attn_implementation="eager"` is required for phi-3-vision
+    # `modules_to_not_convert=["vision_embed_tokens"]` and `model = model.half()` are for acceleration and are optional
+
+    # When running LLMs on Intel iGPUs for Windows users, we recommend setting `cpu_embedding=True` in the from_pretrained function.
+    # This will allow the memory-intensive embedding layer to utilize the CPU instead of iGPU.
+    model = AutoModelForCausalLM.from_pretrained(model_path,
+                                                 trust_remote_code=True,
+                                                 load_in_low_bit="sym_int8",
```

Comment:
shall we also try fp6?

Reply:
Thank you for the comment. We will have a try :)

Reply:
Have some tested and fp6 & fp8 both shows reasonable response. Will use `fp8` as default, and add `fp6` and `fp8` in comments :)

int8
![image](https://github.com/intel-analytics/ipex-llm/assets/54161268/5bd3557d-c8e1-4b85-b76e-db43c066fcc7)
fp6
![image](https://github.com/intel-analytics/ipex-llm/assets/54161268/aff8fb56-58fc-4cbc-a54b-01bfaadfd2f5)
fp8
![image](https://github.com/intel-analytics/ipex-llm/assets/54161268/e9c66d7d-ebef-476a-bc2a-41329a452a83)

https://assets-c4akfrf5b4d3f4b7.z01.azurefd.net/assets/2024/04/BMDataViz_661fb89f3845e.png